modules/tensor_mechanics/include/kernels/ADStressDivergenceShell.h

```diff
@@ -0,0 +1,66 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADKernel.h"
+#include "RankTwoTensor.h"
+#include "RankFourTensor.h"
+#include "libmesh/quadrature_gauss.h"
+
+#define usingADStressDivergenceShellMembers usingKernelMembers;
+
+// Forward Declarations
+template <ComputeStage>
+class ADStressDivergenceShell;
+
+namespace libMesh
+{
+class QGauss;
+}
+
+declareADValidParams(ADStressDivergenceShell);
+
+/**
+ * StressDivergenceTensors mostly copies from StressDivergence.  There are small changes to use
+ * RankFourTensor and RankTwoTensors instead of SymmElasticityTensors and SymmTensors.  This is done
+ * to allow for more mathematical transparancy.
```

Comment:
Thank you Swetha for the nice PR! I do have a question - could you explain a bit on what you are referring to by "mathematical transparency"? I was thinking that for shells we are using the same symmetric elasticity & stress/strain tensors, but maybe you were referring to something else that I didn't think of.

Reply:
Thanks for reviewing the PR, Shawn. This description actually belongs to StressDivergenceTensors class. I forgot to edit the description after making a copy of the file. Thanks for pointing this out. This comment does not apply for shells.

Reply:
Ah OK! Thanks Swetha for explaining.